src/instruments/src/SD/Pages/Hyd/Hyd.tsx

```diff
@@ -209,243 +123,329 @@ const RAT = ({ x, y }: RATProps) => {
+
+    let elecHorizontalLineFormat: string;
+    let verticalLineFormat: string;
+    let elecTriangleFill: number;
+    let elecTriangleColour: string;
+
+    if (!pumpPushbuttonOn) {
+        elecTriangleFill = 0;
+        elecHorizontalLineFormat = 'Hide';
+        elecTriangleColour = 'White';
+    } else {
+        elecTriangleFill = 1;
+        elecHorizontalLineFormat = pressure <= 1450 ? 'AmberLine' : 'GreenLine';
+        elecTriangleColour = pressure <= 1450 ? 'Amber' : 'Green';
+    }
+
+    if ((enginePumpPressureLowSwitch && !pumpPushbuttonOn) || pressure <= 1450) {
+        verticalLineFormat = 'AmberLine';
+    } else {
+        verticalLineFormat = 'GreenLine';
+    }
+
+    return (
+        <>
+            <text
+                id="ELEC-right"
+                className={!ACBus2IsPowered ? 'Large Amber' : 'Large'}
+                x={676}
+                y={292}
+            >
+                ELEC
+            </text>
+            <Triangle x={642} y={283} scale={4 / 3} colour={elecTriangleColour} fill={elecTriangleFill} orientation={-90} />
+            <line className={elecHorizontalLineFormat} x1={631} y1={283} x2={642} y2={283} />
+            <line className={verticalLineFormat} x1={630} y1={217} x2={630} y2={283} />
         </>
     );
 };
 
+enum TransferColor {
+    Green = 'Green',
+    Amber = 'Amber',
+}
+
+enum TransferState {
+    GreenToYellow,
+    YellowToGreen,
+    None
+}
+
 type PTUProps = {
     x: number,
     y: number,
-    ptuScenario: string
+    yellowPressure: number,
+    greenPressure: number,
+    yellowPumpLowPressure: boolean,
+    greenPumpLowPressure: boolean,
+    yellowQuantity: number,
+    greenQuantity: number,
+    ptuControlValveOff: boolean
+    yellowElecPumpOn: boolean,
 }
 
-const PTU = ({ x, y, ptuScenario } : PTUProps) => {
-    const semiCircleD = `M${x - 16},${y} C${x - 16},${y + 24} ${x + 16},${y + 24} ${x + 16},${y}`;
+const shouldTransferActivate = (
+    lowerPressureSystemQuantity: number,
+    lowerPressureSystemPressure: number,
+    highPressureSystemPressure: number,
+    lowerPressureSystemPumpLowPress: boolean,
+    yellowElecPumpOn: boolean,
+    transferDirection: TransferState,
+) => {
+    if (transferDirection === TransferState.None) {
+        return false;
+    }
+
+    const litersPerGallon = 3.79;
+
+    return lowerPressureSystemPumpLowPress
+        && (transferDirection === TransferState.GreenToYellow && !yellowElecPumpOn || transferDirection === TransferState.YellowToGreen)
+        && ((highPressureSystemPressure > 1450 && lowerPressureSystemQuantity * litersPerGallon < 2.5)
+        || (lowerPressureSystemPressure > 1500 && lowerPressureSystemQuantity * litersPerGallon > 2.5));
+};
+
+const PTU = ({ x, y, yellowPressure, greenPressure, yellowPumpLowPressure, greenPumpLowPressure, yellowQuantity, greenQuantity, ptuControlValveOff, yellowElecPumpOn } : PTUProps) => {
+    const [transferColor, setTransferColor] = useState(TransferColor.Green);
```

Comment:
Why does this have to be stored in state? It looks like it only depends on `ptuControlValveOff`.

Reply:
You're right. I originallly thought that it would need to persistent too, but it turned out it doesn't and I forgot to remove this.